rust/connlib/tunnel/src/sockets.rs

```diff
@@ -254,28 +262,40 @@ impl Socket {
         }
     }
 
-    fn poll_send_ready(&self, cx: &mut Context<'_>) -> Poll<io::Result<()>> {
-        self.socket.poll_send_ready(cx)
+    fn poll_flush(&mut self, cx: &mut Context<'_>) -> Poll<io::Result<()>> {
+        loop {
+            match self.socket.try_io(Interest::WRITABLE, || {
```

Comment:
What does `try_io` do that's different from checking if it's writable and then writing? It's not like it avoids a TOCTOU error, right?

Reply:
It internally clears the readiness flag of the socket. As far as I understand, that is what `poll_send_ready` returns.

I.e. we need to clear that flag otherwise it is stale and we never register a waker for when it is ready.